tactics/hints.ml

```diff
@@ -1175,28 +1175,26 @@ let load_autohint _ h =
   | AddMode { gref; mode } ->
     if superglobal then add_mode name gref mode
 
-let open_autohint i h =
+let open_autohint i ((sp,kn), h) =
   let superglobal = superglobal h in
   if Int.equal i 1 then match h.hint_action with
   | AddHints hints ->
-    let () =
-      if not superglobal then
-        (* Import-bound hints must be declared when not imported yet *)
-        let filter (_, h) = not @@ KNmap.mem h.code.uid !statustable in
-        add_hint h.hint_name (List.filter filter hints)
-    in
+    if not superglobal then
+      (* Import-bound hints must be declared when not imported yet *)
+      let filter (_, h) = not @@ KNmap.mem h.code.uid !statustable in
+      add_hint h.db_name (List.filter filter hints);
     let add (_, hint) = statustable := KNmap.add hint.code.uid true !statustable in
     List.iter add hints
   | AddCut paths ->
-    if not superglobal then add_cut h.hint_name paths
+    if not superglobal then add_cut h.db_name paths
   | AddTransparency { grefs; state } ->
-    if not superglobal then add_transparency h.hint_name grefs state
+    if not superglobal then add_transparency h.db_name grefs state
   | RemoveHints hints ->
-    if not superglobal then remove_hint h.hint_name hints
+    if not superglobal then remove_hint h.db_name hints
   | AddMode { gref; mode } ->
-    if not superglobal then add_mode h.hint_name gref mode
+    if not superglobal then add_mode h.db_name gref mode
 
-let cache_autohint o =
+let cache_autohint ((sp,kn), o) =
```

Comment:
```suggestion
let cache_autohint o =
```
this should fix the type error which ocaml locates at `load_function = load_autohint